vllm/logging_utils/redact_filter.py

```diff
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import re
+
+
+class RedactFilter(logging.Filter):
+    """Redacts sensitive information from log messages."""
+
+    def __init__(self, patterns):
+        super().__init__()
+        self._patterns = [re.compile(pattern) for pattern in patterns]
+
+    def filter(self, record):
+        redacted_msg = record.getMessage()
+        for pattern in self._patterns:
+            redacted_msg = pattern.sub("[...]", redacted_msg)
+
+        # Update msg and clear args only if redaction occurred
+        if redacted_msg != record.getMessage():
```

Comment:
this string comparison will be executed even if we have no filters, and will slow low the critical path.

can you add a shortcut for the case where we don't have filters?

Reply:
Thanks for having a look at this. Added a check, skips comparison/filtering if there are no patterns provided:

https://github.com/vllm-project/vllm/pull/13225/files#diff-7737fae74497496304e7635b305eae15e6314ad07d64693cb4b868caf7f898a5R14-R15